### PR TITLE
LG-3312: Fix Combobox Props

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3312](https://jira.mongodb.org/browse/LG-3312)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/combobox/src/Combobox.types.ts [hunk 1]
```diff
@@ -132,141 +133,118 @@ export interface ComboboxMultiselectProps<M extends boolean> {
...
-   * When usePortal is `true`, specifies the scrollable element to position relative to.
-   */
-  scrollContainer?: null;
-
-  /**
-   * Number that controls the z-index of the popover element directly.
-   */
-  popoverZIndex?: number;
-}
+export type BaseComboboxProps = Omit<HTMLElementProps<'div'>, 'onChange'> &
```
Comment (TheSonOfThomp) [inline packages/combobox/src/Combobox.types.ts:136]: nit: I feel like we should keep this an interface
```ts
interface BaseComboboxProps 
  extends Omit<HTMLElementProps<'div'>, 'onChange'>, PortalControlProps {...}
```

File: packages/combobox/src/Combobox.types.ts [hunk 2]
```diff
@@ -132,141 +133,118 @@ export interface ComboboxMultiselectProps<M extends boolean> {
...
+    /**
+     * Defined the character limit of a multiselect Chip before they start truncating.
+     * Note: the three ellipses dots are included in the character limit.
+     */
+    chipCharacterLimit?: number;
+
+    /**
+     * Number that controls the z-index of the popover element directly.
+     */
+    popoverZIndex?: number;
```
Comment (TheSonOfThomp) [inline packages/combobox/src/Combobox.types.ts:246]: is `popoverZIndex` not inherited?

File: packages/combobox/src/utils/ComboboxTestUtils.tsx [hunk 3]
```diff
@@ -199,8 +199,12 @@ export function renderCombobox<T extends Select>(
    * @param newProps
    * @returns
    */
-  const rerenderCombobox = (newProps: renderComboboxProps) =>
-    renderResult.rerender(getComboboxJSX({ ...props, ...newProps }));
+  const rerenderCombobox = (newProps: renderComboboxProps) => {
+    const rerenderProps = { ...props, ...newProps };
+    return renderResult.rerender(
+      getComboboxJSX(rerenderProps as renderComboboxProps),
```
Comment (TheSonOfThomp) [inline packages/combobox/src/utils/ComboboxTestUtils.tsx:205]: We should type the variable not the argument
```ts
const rerenderProps: renderComboboxProps = { ...props, ...newProps };
```

**Review by TheSonOfThomp — APPROVED** [review]

File: packages/combobox/src/Combobox.types.ts (hunk 2 above)
Comment (bruugey) [reply packages/combobox/src/Combobox.types.ts:246]: Not from PopoverControls

File: packages/combobox/src/Combobox.types.ts (hunk 1 above)
Comment (bruugey) [reply packages/combobox/src/Combobox.types.ts:136]: PortalControlProps isn't statically known so we get a TS error

File: packages/combobox/src/utils/ComboboxTestUtils.tsx (hunk 3 above)
Comment (bruugey) [reply packages/combobox/src/utils/ComboboxTestUtils.tsx:205]: we have to cast it, TS still raises an error when we type the variable

Comment (TheSonOfThomp) [reply packages/combobox/src/utils/ComboboxTestUtils.tsx:205]: Hmm, I still think asserting at assignment is technically better
```ts
const rerenderProps = {
    ...props,
    ...newProps,
} as renderComboboxProps;
```

**Pull request merged**